core/txpool/legacypool/legacypool.go

```diff
@@ -1954,3 +1954,22 @@ func (t *lookup) RemotesBelowTip(threshold *big.Int) types.Transactions {
 func numSlots(tx *types.Transaction) int {
 	return int((tx.Size() + txSlotSize - 1) / txSlotSize)
 }
+
+// DropTransactions implements txpool.SubPool, removing all tracked txs from the pool
+// and rotating the journal.
+func (p *LegacyPool) DropTransactions() {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	p.all = newLookup()
```

Comment:
We should probably iterate and unreserve the addresses here. It's less optimal that recreating the reservations map in the parent pool, but that's very sensitive datarace wise, so it's simpler to iterate for now.